buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClusterConfiguration.java

```diff
@@ -46,8 +48,14 @@
 
     void plugin(File plugin);
 
+    void plugin(Supplier<URI> plugin);
```

Comment:
Why do we also need this Supplier variant? Seems inconsistent to have it for plugin but not module.

Reply:
Seems we do this so we can internally keep this as a `List<Supplier<URI>>`. I think we can ditch this and use `Provider<URI>` for the internal collection. Just do `plugin.map(File::toURI)` before adding it to the backing collection.

Reply:
@rjernst / @mark-vieira  thanks - I removed the Supplier variant in favor of just the Gradle Provider (I introduced the supplier because I didn't realize I could create Provider<URI>)

I didn't need to use the `.map` here, but did to implement the conversion from `RegularFile` to `File` via the [other comment](https://github.com/elastic/elasticsearch/pull/54852#discussion_r405054650).